<launch>
  <env name="ROSCONSOLE_CONFIG_FILE" value="$(find formation_control)/config/rosconsole_debug_enabled.conf"/>

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find formation_control)/config/config.rviz" output="screen"/>

  <node pkg="tf" type="static_transform_publisher" name="fixed_frame_pub" args="0.0 0.0 0.0 0.0 0.0 0.0 map map_origin 100" />

  <node pkg="formation_control" type="visualization" name="visualization" output="screen" cwd="ROS_HOME">
    <param name="number_of_agents" type="int" value="5" />
    <rosparam command="load" file="$(find formation_control)/config/visualization_initialization.yaml" />
  </node>

  <arg name="id_1" default="1"/>
  <node pkg="formation_control" type="agent" name="agent_$(arg id_1)" output="screen" cwd="ROS_HOME">
    <param name="agent_id" type="int" value="$(arg id_1)" />
    <rosparam command="load" file="$(find formation_control)/config/agent_initialization.yaml" />
  </node>

  <arg name="id_2" default="2"/>
  <node pkg="formation_control" type="agent" name="agent_$(arg id_2)" output="screen" cwd="ROS_HOME">
    <param name="agent_id" type="int" value="$(arg id_2)" />
    <rosparam command="load" file="$(find formation_control)/config/agent_initialization.yaml" />
  </node>

  <arg name="id_3" default="3"/>
  <node pkg="formation_control" type="agent" name="agent_$(arg id_3)" output="screen" cwd="ROS_HOME">
    <param name="agent_id" type="int" value="$(arg id_3)" />
    <rosparam command="load" file="$(find formation_control)/config/agent_initialization.yaml" />
  </node>

  <arg name="id_4" default="4"/>
  <node pkg="formation_control" type="agent" name="agent_$(arg id_4)" output="screen" cwd="ROS_HOME">
    <param name="agent_id" type="int" value="$(arg id_4)" />
    <rosparam command="load" file="$(find formation_control)/config/agent_initialization.yaml" />
  </node>

  <arg name="id_5" default="5"/>
  <node pkg="formation_control" type="agent" name="agent_$(arg id_5)" output="screen" cwd="ROS_HOME">
    <param name="agent_id" type="int" value="$(arg id_5)" />
    <rosparam command="load" file="$(find formation_control)/config/agent_initialization.yaml" />
  </node>
</launch>
